models/Role.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Role Model
+ *
+ * Defines different roles a user can have (e.g., admin, user, moderator(graveyard keepers)).
+ */
+
+module.exports = (sequelize, Sequelize) => {
+  const { DataTypes } = Sequelize;
+
+  const Role = sequelize.define(
+    "Role",
+    {
+      id: {
+        type: DataTypes.INTEGER,
+        autoIncrement: true,
+        primaryKey: true,
+      },
+      name: {
+        type: DataTypes.STRING,
+        allowNull: false,
+        unique: true,
+      },
+      permissionLevel: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        defaultValue: 1,
+        comment: "Numeric access level representing permission hierarchy (higher = more access)",
+      },
+    },
+    {
+      timestamps: true,
+      tableName: "Roles",
+    }
+  );
+
+  Role.associate = (models) => {
+    Role.hasMany(models.User, {
+      foreignKey: "roleId",
+      as: "users",
+      onDelete: "SET NULL",
```

Comment:
When deleting a role I would avoid setting a users role to null as they would be without permissions technically. 

```suggestion
     onDelete: "RESTRICT",
```